### ISSUE TITLE
configure in python directory can't use mpfr_complex.hpp

### ISSUE THREAD

**Issue by jacobbond** [bertiniteam/b2#48]
configure: found bertini2 includedir
configure: found bertini2 libdir
checking bertini2/mpfr_complex.hpp usability... no
checking bertini2/mpfr_complex.hpp presence... yes
configure: WARNING: bertini2/mpfr_complex.hpp: present but cannot be compiled
configure: WARNING: bertini2/mpfr_complex.hpp:     check for missing prerequisite headers?
configure: WARNING: bertini2/mpfr_complex.hpp: see the Autoconf documentation
configure: WARNING: bertini2/mpfr_complex.hpp:     section "Present But Cannot Be Compiled"
configure: WARNING: bertini2/mpfr_complex.hpp: proceeding with the compiler's result
configure: WARNING:     ## --------------------------------------- ##
configure: WARNING:     ## Report this to  ##
configure: WARNING:     ## --------------------------------------- ##
checking for bertini2/mpfr_complex.hpp... no
configure: error: unable to find required file mpfr_complex.hpp in include for Bertini2

Bertini was installed to a custom location using `configure`'s `--prefix` flag and then `--with-bertini` was set for the `configure` in the `python` directory.

I just tried installing to the default location (`/usr/local`; no `--prefix` or `--with-bertini` flags on `configure`) and I got the same error.

Is it possible that some of the Boost flags (`BOOST_CPPFLAGS` or `BOOST_LDFLAGS`) aren't being passed to the test compilation?

**Comment by jacobbond** [comment]
Bertini was installed to a custom location using `configure`'s `--prefix` flag and then `--with-bertini` was set for the `configure` in the `python` directory.

**Comment by jacobbond** [comment]
I just tried installing to the default location (`/usr/local`) and I got the same error.

**Comment by ofloveandhate** [comment]
in `config.log`, what does it say?  does it indicate a missing header file?

**Comment by ofloveandhate** [comment]
or attach config.log to let me help determine.

**Comment by ofloveandhate** [comment]
and just to be clear, you are `autoreconf`ing before retrying configuring, after pulling a set of commits, right?

**Comment by jacobbond** [comment]
Yeah, missing headers.  It never got `BOOST_CPPFLAGS` (there's no `-I/sd/SageMath/local/include`).  I attached it.
[config.log.txt](https://github.com/bertiniteam/b2/files/447571/config.log.txt)

**Comment by jacobbond** [comment]
Yeah, I'm `autoreconf`ing.

**Comment by ofloveandhate** [comment]
ok, you're right, the `BOOST_CPPFLAGS` didn't make in into the command to test compiling with mpfr_complex.hpp.    the relevant lines are

```
configure:19178: g++ -c -g -O2 -std=gnu++14  -I/sd/eigen -I/usr/local/include conftest.cpp >&5
In file included from /usr/local/include/bertini2/mpfr_complex.hpp:37:0,
                 from conftest.cpp:70:
/usr/local/include/bertini2/mpfr_extensions.hpp:186:30: error: 'boost::multiprecision::min' has not been declared
 using boost::multiprecision::min; //error receiver: please see https://svn.boost.org/trac/boost/ticket/11149 for information about these using statements in std namespace.  
                              ^
In file included from /usr/local/include/bertini2/mpfr_complex.hpp:37:0,
                 from conftest.cpp:70:
/usr/local/include/bertini2/mpfr_extensions.hpp:187:30: error: 'boost::multiprecision::max' has not been declared
 using boost::multiprecision::max; //3 options: ./configure --disable-expression_templates, use Boost 1.61, or patch earlier Boost versions to resolve this.
```

i think it is finding an earlier version of boost since the CPPFLAGS isn't correct.  i am guessing you have something earlier than 1.61 installed somewhere else on this computer?  not a problem, i will determine how to get BOOST_CPPFLAGS into AX_BERTINI2.

**Comment by ofloveandhate** [comment]
I note that if an autotools or m4 expert reads this, we could use help writing this macro...

**Comment by ofloveandhate** [comment]
new commit 
https://github.com/bertiniteam/b2/commit/ff49b4a5abf86b42467b80f1a7859d7683c7ca58
adds boostcppflags when finding bertini2.  give it a try?

**Comment by jacobbond** [comment]
When it is compiling `conftest`, it fails and it looks like it's because it has the wrong value for `BERTINI_LDFLAGS`.  By default, Bertini is installed into `/usr/local/include` and `/usr/local/lib`, but `BERTINI_LDFLAGS` is set to `/usr/lib`.
[config.log.txt](https://github.com/bertiniteam/b2/files/447722/config.log.txt)

**Comment by jacobbond** [comment]
Actually, I don't know if it's the value of `BERTINI_LDFLAGS` that is the issue.  Even if it's set correctly, it still fails the same way
[config.log.txt](https://github.com/bertiniteam/b2/files/447767/config.log.txt)
:

`configure:19228: g++ -o conftest -g -O2 -std=gnu++14  -I/sd/eigen -pthread -I/sd/SageMath/local/include -I/sd/bertini/include  -L/sd/bertini/lib conftest.cpp -lmpfr -lgmp  >&5`

`configure:19228: g++ -o conftest -g -O2 -std=gnu++14  -I/sd/eigen -pthread -I/sd/SageMath/local/include -I/sd/bertini/include  -L/sd/bertini/lib conftest.cpp -lbertini2 -L/sd/SageMath/local/lib
        -lmpfr -lgmp  >&5`

**Comment by ofloveandhate** [comment]
i have been able to replicate your error on my ubuntu machine.

**Comment by ofloveandhate** [comment]
in your most recent message, the fail is due to not linking to boost libraries when linking to bertini2.

**Comment by ofloveandhate** [comment]
well, i've got it finding libbertini2, but the linking still fails when testing for libbertini2 due to a boost.serialization definition missing...  argh.

**Comment by jacobbond** [comment]
Configure worked just fine in the last commit.

**Comment by ofloveandhate** [comment]
this was fixed in #50
